include/deal.II/non_matching/mapping_info.h

```diff
@@ -57,27 +62,125 @@ namespace NonMatching
     MappingInfo(const Mapping<dim> &mapping, const UpdateFlags update_flags);
 
     /**
-     * Reinitialize the mapping information for the incoming cell and unit
+     * Compute the mapping information for the incoming cell and unit
+     * points. (needed to resolve ambiguity)
+     */
+    void
+    reinit(const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+           const std::vector<Point<dim>> &unit_points);
+
+    /**
+     * Compute the mapping information for the incoming cell and unit
      * points.
      */
     void
     reinit(const typename Triangulation<dim, spacedim>::cell_iterator &cell,
            const ArrayView<const Point<dim>> &unit_points);
 
+    /**
+     * Compute the mapping information for the given cell and
+     * quadrature formula. As opposed to the other `reinit` function, this
+     * method allows to access a `JxW` factor at the points.
+     */
+    void
+    reinit(const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+           const Quadrature<dim> &quadrature);
+
+    /**
+     * Compute the mapping information for the incoming vector of cells and
+     * corresponding vector of unit points.
+     *
+     * It is possible to give an IteratorRange<FilteredIterator> to this
+     * function and together with @p n_unfiltered_cells specified this object
+     * can compress its storage while giving you access to the underlying data
+     * with the "unfiltered" index. The default argument for
+     * @p n_unfiltered_cells disables this built-in compression.
```

Comment:
I don't get how this should work...

Reply:
I updated the test to show how to use this feature.